src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +24,38 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    private static final int MAX_COUNT_OF_LETTERS = 5;
+
+    public String getUniqueCharacters(String fileName) throws IOException {
+        String line;
+        StringBuilder concat = new StringBuilder();
+        Set<Character> set = new TreeSet<>();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            while ((line = reader.readLine()) != null) {
+                concat.append(line.toLowerCase());
```

Comment:
There is a way, you can check each character right here and if it is OK, add to Set